tracee-rules/engine/engine_test.go

```diff
@@ -0,0 +1,238 @@
+package engine
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/aquasecurity/tracee/tracee-rules/types"
+	"github.com/stretchr/testify/assert"
+)
+
+type fakeSignature struct {
+	getMetadata       func() (types.SignatureMetadata, error)
+	getSelectedEvents func() ([]types.SignatureEventSelector, error)
+	init              func(types.SignatureHandler) error
+	onEvent           func(types.Event) error
+	onSignal          func(signal types.Signal) error
+}
+
+func (fs fakeSignature) GetMetadata() (types.SignatureMetadata, error) {
+	if fs.getMetadata != nil {
+		return fs.getMetadata()
+	}
+
+	return types.SignatureMetadata{
+		Name: "Fake Signature",
+	}, nil
+}
+
+func (fs fakeSignature) GetSelectedEvents() ([]types.SignatureEventSelector, error) {
+	if fs.getSelectedEvents != nil {
+		return fs.getSelectedEvents()
+	}
+
+	return []types.SignatureEventSelector{}, nil
+}
+
+func (fs fakeSignature) Init(cb types.SignatureHandler) error {
+	if fs.init != nil {
+		return fs.init(cb)
+	}
+	return nil
+}
+
+func (fs fakeSignature) OnEvent(event types.Event) error {
+	if fs.onEvent != nil {
+		return fs.onEvent(event)
+	}
+	return nil
+}
+
+func (fs fakeSignature) OnSignal(signal types.Signal) error {
+	if fs.onSignal != nil {
+		return fs.onSignal(signal)
+	}
+	return nil
+}
+
+func TestEngine_consumeSource(t *testing.T) {
```

Comment:
Shouldn't it be called TestEngineConsumeSource?

Reply:
actually it should be `TestConsumeSources`. engine is the package and file being tested

Reply:
Updated https://github.com/aquasecurity/tracee/pull/477/commits/2cd3c45e11a781604fb0d5d261cc31e7d420777a